boat_simulator/nodes/physics_engine/fluid_forces.py

```diff
@@ -0,0 +1,76 @@
+"""This module provides functionality for computing the lift and drag forces acting on a medium."""
+
+from typing import Dict, Tuple
+
+import numpy as np
+from numpy.typing import NDArray
+
+from boat_simulator.common.types import Scalar
+
+
+class MediumForceComputation:
+    """This class calculates the lift and drag forces experienced by a medium when subjected to
+    fluid flow.
+
+    Attributes:
+        `lift_coefficients` (Dict[Scalar, Scalar]): A dictionary mapping angles of attack to lift
+            coefficients. Keys represent different angles of attack in degrees, and values
+            represent corresponding lift coefficients.
+        `drag_coefficients` (Dict[Scalar, Scalar]): A dictionary mapping angles of attack to drag
+            coefficients. Keys represent different angles of attack in degrees, and values
+            represent corresponding drag coefficients.
+        `areas` (Dict[Scalar, Scalar]): A dictionary mapping angles of attack to the area of the
+            medium exposed to fluid flow. Keys represent different angles of attack in degrees, and
+            values represent the corresponding areas in square meters (m^2).
+        `fluid_density` (Scalar): The density of the fluid acting on the medium, expressed in
+            kilograms per cubic meter (kg/m^3).
+    """
+
+    def __init__(
+        self,
+        lift_coefficients: Dict[Scalar, Scalar],
+        drag_coefficients: Dict[Scalar, Scalar],
+        areas: Dict[Scalar, Scalar],
```

Comment:
There might be problems when hashing with a float when using dictionaries, so I recommend using a 2D array instead. In particular, I recommend an array with shape `(n,2)` there the `(x,y)` pairs are in each row (this kinda enforces the shape almost because with a `(2,n)` array, you need to know that the shapes match ahead of time).

These arrays would be used in an interpolation function since the angle of attacks that we compute will very likely not fall on exactly any of the values that we have here. @alberto-escobar is working on a lookup table class, so maybe we could use that to help with the implementation here.

However, this is not finished so perhaps we could just have a simple linear interpolation function inside this class that uses `np.interp` so we can compute the coefficients given an angle of attack